samples/extensions/khr/nbody/main.cpp

```diff
@@ -155,10 +155,8 @@ void NBody::initializeGL()
     if (glewInit() != GLEW_OK) std::exit(EXIT_FAILURE);
 
     auto create_shader = [](std::string file_path, cl_GLenum shader_stage) {
-        std::ifstream shader_stream(file_path);
-        std::string shader_string{ std::istreambuf_iterator<GLchar>{
-                                       shader_stream },
-                                   std::istreambuf_iterator<GLchar>{} };
+        std::string shader_string =
+            cl::util::read_exe_relative_text_file(file_path.c_str());
```

Comment:
Add an overload taking a `std::string`. Actually a `string_view` would be even better when it is available.

Reply:
I would like to make that a separate discussion, because some of the overloads would need to be version guarded.

Reply:
I agree it would be cleaner if the C++ version had a `const std::string&` overload, which shouldn't need any version guards.  This could be done in a subsequent PR, though.